src/core/__tests__/ReactCompositeComponent-test.js

```diff
@@ -599,6 +599,128 @@ describe('ReactCompositeComponent', function() {
 
   });
 
+  it('should warn if context values differ on update', function() {
+    var Parent = React.createClass({
+      childContextTypes: {
+        foo: ReactPropTypes.string
+      },
+
+      getChildContext: function() {
+        return {
+          foo: "bar"
+        };
+      },
+
+      render: function() {
+        return <div>{this.props.children}</div>;
+      }
+    });
```

Comment:
Nit: Speaking of code formatting. Personally I tend to prefer to have an extra new line between these. Otherwise it seems like everything from render to the end of contextTypes belong together. Subjectively I look for spaces as separators.